canopen/objectdictionary/eds.py

```diff
@@ -19,19 +19,17 @@
 def import_eds(source, node_id):
     eds = RawConfigParser(inline_comment_prefixes=(';',))
     eds.optionxform = str
-    if hasattr(source, "read"):
-        fp = source
-    else:
-        fp = open(source)
+    opened_here = False
     try:
-        # Python 3
+        if hasattr(source, "read"):
+            fp = source
+        else:
+            fp = open(source)
+            opened_here = True
         eds.read_file(fp)
-    except AttributeError:
-        # Python 2
-        eds.readfp(fp)
     finally:
         # Only close object if opened in this fn
-        if not hasattr(source, "read"):
+        if opened_here:
```

Comment:
The boolean flag seems like a good idea.  But I don't see why anything before the `open()` call should be enclosed in the `try ... finally` block?

Reply:
How would you do it without?

```python
if hasattr(source, "read"):
    fp = source
try:
    if not hasattr(source, "read"):
        fp = open(source)
        opened_here = True
    eds.read_file(fp)
finally:
    ...
````

This is the same line count and a little less DRY, right?

Reply:
Sorry, you're right. It doesn't make much sense to avoid that. I probably misread the diff.